http/server.lua

```diff
@@ -685,21 +729,22 @@ local function process_client(self, s, peer)
             s:write('HTTP/1.0 100 Continue\r\n\r\n')
         end
 
-        local logreq = self.options.log_requests and log.info or log.debug
+        local route = self:match(p.method, p.path)
+        local logreq = get_request_logger(self.options, route)
         logreq("%s %s%s", p.method, p.path,
-            p.query ~= "" and "?"..p.query or "")
+               p.query ~= "" and "?"..p.query or "")
 
-        local res, reason = pcall(self.options.handler, self, p)
+        local res, reason = pcall(self.options.handler, self, route, p)
```

Comment:
Maybe better leave double-dispatch and not change router signature?

Reply:
Yes, it will be more obvious